# HG changeset patch
# User Rik <rik@octave.org>
# Date 1344037517 25200
#      Fri Aug 03 16:45:17 2012 -0700
# Node ID 8523df595d42d46603aabd1bdb7371e16c29bebe
# Parent  d7f8b0b91cb2033fab7625cc015b9e84d336d85a
maint: Add DO NOT EDIT messages to build system for src/*.in.h.

* build-aux/common.mk (do_subst_config_val, do_subst_default_val):
Add new sed expansion rule for NO_EDIT_WARNING.

* Makefile.am (version.h rule):
Add new sed expansion rule for NO_EDIT_WARNING.

* interp-core/module.mk (mxarray.h):
Add new sed expansion rule for NO_EDIT_WARNING.

* mxarray.in.h, defaults.in.h, oct-conf.in.h, version.in.h:
Add %NO_EDIT_WARNING% to top-of-file for expansion by build system.

diff --git a/build-aux/common.mk b/build-aux/common.mk
--- a/build-aux/common.mk
+++ b/build-aux/common.mk
@@ -449,16 +449,17 @@ endef
 
 # Yes, the second sed command near the end is needed, to avoid limits
 # in command lengths for some versions of sed.  UGLY_DEFS is often
 # quite large, so it makes sense to split this command there.
 
 define do_subst_config_vals
 echo "making $@ from $<"
 $(SED) < $< \
+  -e "s|%NO_EDIT_WARNING%|DO NOT EDIT!  Generated automatically from $(<F) by Make.|" \
   -e "s|%NO_OCT_FILE_STRIP%|${NO_OCT_FILE_STRIP}|" \
   -e "s|%OCTAVE_BINDIR%|\"${bindir}\"|" \
   -e "s|%OCTAVE_CONF_ALL_CFLAGS%|\"${ALL_CFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_ALL_CXXFLAGS%|\"${ALL_CXXFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_ALL_FFLAGS%|\"${ALL_FFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_ALL_LDFLAGS%|\"${ALL_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_AMD_CPPFLAGS%|\"${AMD_CPPFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_AMD_LDFLAGS%|\"${AMD_LDFLAGS}\"|" \
@@ -601,16 +602,17 @@ echo "making $@ from $<"
   -e "s|%OCTAVE_CONF_config_opts%|\"${config_opts}\"|" | \
   $(SED)  -e "s|%OCTAVE_CONF_DEFS%|\"${UGLY_DEFS}\"|" > $@-t
 $(simple_move_if_change_rule)
 endef
 
 define do_subst_default_vals
 echo "making $@ from $<"
 $(SED) < $< > $@-t \
+  -e "s|%NO_EDIT_WARNING%|DO NOT EDIT!  Generated automatically from $(<F) by Make.|" \
   -e "s|%OCTAVE_ARCHLIBDIR%|\"${archlibdir}\"|" \
   -e "s|%OCTAVE_BINDIR%|\"${bindir}\"|" \
   -e "s|%OCTAVE_CANONICAL_HOST_TYPE%|\"${canonical_host_type}\"|" \
   -e "s|%OCTAVE_DATADIR%|\"${datadir}\"|" \
   -e "s|%OCTAVE_DATAROOTDIR%|\"${datarootdir}\"|" \
   -e "s|%OCTAVE_DEFAULT_PAGER%|\"${DEFAULT_PAGER}\"|" \
   -e "s|%OCTAVE_DOC_CACHE_FILE%|\"${doc_cache_file}\"|" \
   -e "s|%OCTAVE_EXEC_PREFIX%|\"${exec_prefix}\"|" \
diff --git a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -259,16 +259,17 @@ DLL_CXXDEFS = @OCTINTERP_DLL_DEFS@
 ## oct-conf.h must depend on Makefile.  Calling configure
 ## may change default/config values.  However, calling configure will also
 ## regenerate the Makefiles from Makefile.am and trigger the rules below.
 oct-conf.h: oct-conf.in.h Makefile
 	@$(do_subst_config_vals)
 
 version.h: version.in.h Makefile
 	$(SED) < $< \
+	  -e "s|%NO_EDIT_WARNING%|DO NOT EDIT!  Generated automatically from $(<F) by Make.|" \
 	  -e "s|%OCTAVE_API_VERSION_NUMBER%|${OCTAVE_API_VERSION_NUMBER}|" \
 	  -e "s|%OCTAVE_API_VERSION%|\"${OCTAVE_API_VERSION}\"|" \
 	  -e "s|%OCTAVE_COPYRIGHT%|\"${OCTAVE_COPYRIGHT}\"|" \
 	  -e "s|%OCTAVE_RELEASE_DATE%|\"${OCTAVE_RELEASE_DATE}\"|" \
 	  -e "s|%OCTAVE_VERSION%|\"${OCTAVE_VERSION}\"|" > $@-t
 	mv $@-t $@
 
 builtins.cc: $(DEF_FILES) mkbuiltins
diff --git a/src/interp-core/module.mk b/src/interp-core/module.mk
--- a/src/interp-core/module.mk
+++ b/src/interp-core/module.mk
@@ -120,11 +120,12 @@ interp-core/oct-errno.cc: interp-core/oc
 	  $(srcdir)/mk-errno-list --python "$(PYTHON)" < $< > $@-t; \
 	else \
 	  $(SED) '/@SYSDEP_ERRNO_LIST@/D' $< > $@-t; \
 	fi
 	mv $@-t $@
 
 interp-core/mxarray.h: interp-core/mxarray.in.h Makefile
 	$(SED) < $< \
+	  -e "s|%NO_EDIT_WARNING%|DO NOT EDIT!  Generated automatically from $(<F) by Make.|" \
 	  -e "s|%OCTAVE_IDX_TYPE%|${OCTAVE_IDX_TYPE}|" > $@-t
 	mv $@-t $@
 
diff --git a/src/interp-core/mxarray.in.h b/src/interp-core/mxarray.in.h
--- a/src/interp-core/mxarray.in.h
+++ b/src/interp-core/mxarray.in.h
@@ -1,9 +1,9 @@
-// DO NOT EDIT!  Generated automatically from mxarray.in.h by configure
+// %NO_EDIT_WARNING%
 /*
 
 Copyright (C) 2001-2012 Paul Kienzle
 
 This file is part of Octave.
 
 Octave is free software; you can redistribute it and/or modify it
 under the terms of the GNU General Public License as published by the
diff --git a/src/interpfcn/defaults.in.h b/src/interpfcn/defaults.in.h
--- a/src/interpfcn/defaults.in.h
+++ b/src/interpfcn/defaults.in.h
@@ -1,9 +1,9 @@
-// DO NOT EDIT!  Generated automatically from defaults.in.h by configure
+// %NO_EDIT_WARNING%
 /*
 
 Copyright (C) 1993-2012 John W. Eaton
 
 This file is part of Octave.
 
 Octave is free software; you can redistribute it and/or modify it
 under the terms of the GNU General Public License as published by the
diff --git a/src/oct-conf.in.h b/src/oct-conf.in.h
--- a/src/oct-conf.in.h
+++ b/src/oct-conf.in.h
@@ -1,9 +1,9 @@
-// DO NOT EDIT!  Generated automatically from oct-conf.in.h by configure
+// %NO_EDIT_WARNING%
 /*
 
 Copyright (C) 1996-2012 John W. Eaton
 
 This file is part of Octave.
 
 Octave is free software; you can redistribute it and/or modify it
 under the terms of the GNU General Public License as published by the
diff --git a/src/version.in.h b/src/version.in.h
--- a/src/version.in.h
+++ b/src/version.in.h
@@ -1,9 +1,9 @@
-// DO NOT EDIT!  Generated automatically from version.in.h by configure
+// %NO_EDIT_WARNING%
 /*
 
 Copyright (C) 1992-2012 John W. Eaton
 
 This file is part of Octave.
 
 Octave is free software; you can redistribute it and/or modify it
 under the terms of the GNU General Public License as published by the
